Ku 4- создал телеграм бота, добавил сохранение пользователей в БД, добавил обработку команды /start, поднял проект в докере

KornilovVl opened the pull request:
Создал телеграм- бота
Добавил функционал сохранения пользователей в БД
Добавил обработку команды /start из телеграм-бота
Добавил миграцию для БД
Добавил функицию чтения конфига чтоб уменьшить количество дублирующегося кода
Настроил поднятие проекта в Docker

KornilovVl committed:
Обновил docker-compose.yml и local.toml соответственно
Уточнил в ридми комаду для запуска проекта.
Заменил временно неиспользуемую переменную bd на _ в cmd/api/main.go

KornilovVl committed:
Обновил Readme
Написал миграцию для создания таблицы
Создал модель юзера и БД
Вынес чтение конфигов в отдельный класс
Добавил логику бота для /start

KornilovVl committed:
Обновил Readme
Обновил docker-compose.yml
Изменил файловую структуру проекта

KornilovVl commented:
![](https://github.trello.services/images/mini-trello-icon.png) [KU-4 Подключить бота в микросервис](https://trello.com/c/adouqOIx/7-ku-4-%D0%BF%D0%BE%D0%B4%D0%BA%D0%BB%D1%8E%D1%87%D0%B8%D1%82%D1%8C-%D0%B1%D0%BE%D1%82%D0%B0-%D0%B2-%D0%BC%D0%B8%D0%BA%D1%80%D0%BE%D1%81%D0%B5%D1%80%D0%B2%D0%B8%D1%81)

tungatarovKl commented on cmd/bot/main.go:
```diff
@@ -1,7 +1,26 @@
+)
 
+func main() {
+	println("BOT")
```
дебаг инфо

tungatarovKl commented on cmd/bot/main.go:
```diff
@@ -1,7 +1,26 @@
+	if database.Connection.Error != nil {
+		log.Fatal("Error connecting to database:", database.Connection.Error)
+	}
+	tgBot := Bot.Bot{
```
нужно добавить пустую строку

tungatarovKl commented on cmd/bot/main.go:
```diff
@@ -1,7 +1,26 @@
+		Bot:      Bot.InitBot(cfg.BotToken),
+		Database: database,
+	}
+	tgBot.Bot.Handle("/start", tgBot.StartHandler)
```
нужно добавить пустую строку

tungatarovKl commented on internal/models/migrations/CreateUsersTable.go:
```diff
@@ -0,0 +1,31 @@
+package main
```
использовать package main не в main вводить в заблуждение

KornilovVl committed:
Избавился от миграйций GORM
Добавил SQL файл с миграцией
Добавил проверку на пустые поля в ConfigReader.go

KornilovVl committed:
Внес исправления по комментам

wildb-ee commented on config/ConfigReader.go:
```diff
@@ -0,0 +1,30 @@
+func ReadConfig(configPath string) *Config {
+	var cfg Config
+	if _, err := toml.DecodeFile(configPath, &cfg); err != nil {
+		log.Fatal("Error decoding config: ", configPath, err.Error())
```
Если придираться , я бы перевел все ошибки либо на русский, либо целиком на английский. (Это , конечно, можно сделать когда мы потом будем в конце шлифовать код)

wildb-ee commented on config/ConfigReader.go:
```diff
@@ -0,0 +1,30 @@
+	}
+	if cfg.DbAddress == "" ||
+		cfg.DbName == "" ||
+		cfg.DbPassword == "" ||
```
Если не ошибаюсь, бывают моменты, когда пароль равен "". Тогда это не совсем будет работать (По моему эту ошибку породил еще мой коммит)

KornilovVl committed:
Перевел все ошибки на английский язык,
Добавил возвращение конструктора параметров и ошибки в ConfigReader.go . Теперь ошибку можно игнорировать.
Переделал обработку ошибки подключения к БД
